src/utils/ReconnectManager.js

```diff
@@ -37,74 +37,80 @@ export class ReconnectManager extends EventEmitter {
             return;
         }
 
-        // The threshold for trying glitch-free reconnect should be less than server-side configuration
-        const RECONNECT_THRESHOLD = payload.disconnectTimeout * 0.8;
-        if (Date.now() - (this._signalDisconnectTime || 0) > RECONNECT_THRESHOLD) {
+        // Try to remove our own pending client if this is a page reload
+        // Could also be a first normal room_joined which can never be glitch-free
+        if (!this._signalDisconnectTime) {
+            this._resetClientState(payload);
+            const myDeviceId = payload.room.clients.find((c) => payload.selfId === c.id).deviceId;
+            payload.room.clients = payload.room.clients.filter(
+                (c) => !(c.deviceId === myDeviceId && c.isPendingToLeave)
+            );
             this.emit(PROTOCOL_RESPONSES.ROOM_JOINED, payload);
-
-            // This is either the first connect or a too long disconnect
-            // Reset client-state and add all remote-clients
-            this._clients = {};
-            payload.room.clients.forEach((newClient) => {
-                if (newClient.id === payload.selfId) {
-                    return;
-                } else {
-                    this._addClientToState(newClient);
-                }
-            });
             return;
         }
 
-        const allStats = await getUpdatedStats();
+        // The threshold for trying glitch-free reconnect should be less than server-side configuration
+        const RECONNECT_THRESHOLD = payload.disconnectTimeout * 0.8;
+        if (this._signalDisconnectTime) {
+            if (Date.now() - this._signalDisconnectTime > RECONNECT_THRESHOLD) {
```

Comment:
could we extract out a variable to make it easier to read?

```suggestion
            const timeSinceDisconnect = Date.now() - this._signalDisconnectTime;
            if (timeSinceDisconnect > RECONNECT_THRESHOLD) {
```

Reply:
fixed in https://github.com/whereby/jslib-media/pull/48/commits/6c805ea3585dcbc79d9cd9f86228ccaeaaf10270